h/services/search_index/_queue.py

```diff
@@ -28,78 +33,67 @@ def __init__(self, db, es, batch_indexer):
         self._es = es
         self._batch_indexer = batch_indexer
 
-    def add(self, annotation_id, tag, schedule_in=None, force=False):
+    def add_where(self, tag, priority, where, force=False, schedule_in=None):
         """
-        Queue an annotation to be synced to Elasticsearch.
-
-        :param annotation_id: The ID of the annotation to be queued, in the
-            application-level URL-safe format
-        :type annotation_id: unicode
+        Queue annotations matching a filter to be synced to ElasticSearch.
 
         :param tag: The tag to add to the job on the queue. For documentation
             purposes only
-        :type tag: unicode
-
+        :param priority: Integer priority value (higher number is lower
+            priority)
+        :param where: A list of SQLAlchemy BinaryExpression objects to limit
+            the annotations to be added
+        :param force: Whether to force reindexing of the annotation even if
+            it's already indexed
         :param schedule_in: A number of seconds from now to wait before making
             the job available for processing. The annotation won't be synced
             until at least `schedule_in` seconds from now
-        :type schedule_in: int
-
-        :param force: Whether to force reindexing of the annotation even if
-            it's already indexed
-        :type force: bool
         """
-        self.add_all([annotation_id], tag, schedule_in, force)
+        where_clause = and_(*where) if len(where) > 1 else where[0]
+        schedule_at = datetime.utcnow() + timedelta(seconds=schedule_in or 0)
+
+        query = Job.__table__.insert().from_select(
+            [Job.name, Job.tag, Job.priority, Job.kwargs, Job.scheduled_at],
+            select(
+                [
+                    text("'sync_annotation'"),
+                    text(repr(tag)),
+                    text(str(priority)),
+                    func.jsonb_build_object(
+                        "annotation_id", Annotation.id, "force", bool(force)
```

Comment:
Behavior change here. Added `bool()` around force to prevent us from putting in any old Python stuff into Postgres. Should make querying more reliable if we ever wanted to.